backend/src/bundles/users/user.repository.ts

```diff
@@ -149,12 +192,54 @@ class UserRepository implements Repository {
                 weight: userDetails.weight,
                 height: userDetails.height,
                 gender: userDetails.gender,
+                referralCode: userDetails.referralCode,
+                bonusBalance: userDetails.bonusBalance,
             });
         } catch (error) {
             await trx.rollback();
             throw new Error(`Error updating user details: ${error}`);
         }
     }
+
+    public async createUserBonusTransaction(
+        entity: UserBonusEntity,
+    ): Promise<UserBonusGetAllItemResponseDto | null> {
+        const { userId, actionType, transactionType, amount } =
+            entity.toNewObject();
+        const trx = await this.userModel.startTransaction();
+
+        try {
+            const user = await this.userModel.query(trx).findById(userId);
+
+            if (!user) {
+                return null;
```

Comment:
seems like with such approach transaction will be closed by timeout and before this time there will be open connection to DB, probably better to throw error to close transaction immediately